src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -1223,6 +1245,120 @@ export default class SourceFileLinter {
+						this.#analyzeModelDataTypes(prop.initializer);
+					} else {
+						this.#analyzeModelDataTypesBinding(prop);
+					}
+				}
+			});
+	}
+
+	#analyzeModelDataTypes(node: ts.ObjectLiteralExpression) {
+		node?.properties.forEach((prop) => {
+			if (!ts.isPropertyAssignment(prop)) {
+				return;
+			}
+
+			// Get the type property
+			let typeField;
+			if (ts.isObjectLiteralExpression(prop.initializer)) {
+				typeField = prop.initializer.properties.find((prop: ts.ObjectLiteralElementLike) => {
+					return ts.isPropertyAssignment(prop) &&
+						ts.isIdentifier(prop.name) &&
+						prop.name.text === "type";
+				});
+			} else if (ts.isStringLiteral(prop.initializer) &&
+				ts.isIdentifier(prop.name) && prop.name.text === "type" &&
+				// Whether it's a direct property, named "type" of the Control
+				// or type in property binding
+				!ts.isNewExpression(prop.parent.parent)) {
+				typeField = prop;
+			}
+
+			if (typeField && ts.isPropertyAssignment(typeField)) {
+				this.#analyzeModelTypeField(typeField.initializer);
+			}
+		});
+	}
+
+	#analyzeModelDataTypesBinding(node: ts.PropertyAssignment) {
+		if (ts.isStringLiteral(node.initializer) &&
+			node.initializer.text.startsWith("{") && node.initializer.text.endsWith("}")) {
+			const imports = this.sourceFile.statements
+				.filter((stmnt): stmnt is ts.ImportDeclaration =>
+					stmnt.kind === ts.SyntaxKind.ImportDeclaration)
+				.map((importNode) => {
+					return {
+						moduleName: (importNode.moduleSpecifier as ts.StringLiteral).text,
+						variableName: importNode.importClause?.name?.text,
+					} as RequireDeclaration;
+				});
+
+			const {start: nodePos} = getPositionsForNode({
+				node: node,
+				sourceFile: this.sourceFile,
+				resourcePath: this.resourcePath,
+			});
+			const bindingLinter = new BindingLinter(this.resourcePath, this.context);
+			bindingLinter.lintPropertyBinding(node.initializer.text, imports, nodePos);
+		}
+	}
+
+	/**
+	 * Identify if a constructor argument property is bindable
+	 *
+	 * Get the class declaration by going up the chain and then find the argument property
+	 * from there. Directly finding the type of the property is not possible from here as it's
+	 * missing some context.
+	*/
+	#isPropertyBindingType(node: ts.NewExpression, propName: string) {
+		const controlAmbientModule =
+			this.getSymbolModuleDeclaration(this.checker.getTypeAtLocation(node).symbol);
+		const classArg = (controlAmbientModule?.body as ts.ModuleBlock)?.statements
+			?.find((stmnt): stmnt is ts.ClassDeclaration => stmnt.kind === ts.SyntaxKind.ClassDeclaration)
+			?.members.find((m): m is ts.ConstructorDeclaration => ts.isConstructorDeclaration(m));
+
+		const constructorArgType = classArg && this.checker.getTypeAtLocation(classArg.parameters[0]);
+		const argProperty = constructorArgType?.getProperties()
+			.find((p: ts.Symbol) => p.name === propName);
+
+		const argPropType = argProperty?.declarations?.[0] &&
```

Comment:
This exact declaration of a property within a NewExpression is already available in the existing iteration over the constructor arguments in `analyzeNewExpression`.

The `argPropType` can be accessed there via `this.checker.getTypeAtLocation(propertySymbol.valueDeclaration)`.

Reply:
Thanks!

Reply:
I actually wanted to motivate the re-use of that code so that no additional loop over the arguments and properties is needed at this place. But this can of course also be improved later on.

Reply:
I'm not quite sure how to reuse it. This implementation relies on ambient modules and their definitions.
Trying directly to find the type of an eventual propertyBinding does not provide enough context, so the union types and aliases cannot be determined correctly.